app/NSObject+SaneKVO.m

```diff
@@ -0,0 +1,72 @@
+//
+//  NSObject+SaneKVO.m
+//  iSH
+//
+//  Created by Theodore Dubois on 11/10/20.
+//
+
+#import <objc/runtime.h>
+#import "NSObject+SaneKVO.h"
+
+static void *kKVOObservations = &kKVOObservations;
+
+@interface KVOObservation ()
+- (instancetype)initWithKeyPath:(NSString *)keyPath object:(id)object block:(KVOBlock)block;
+@end
+
+@implementation NSObject (SaneKVO)
+
+- (KVOObservation *)observe:(NSString *)keyPath options:(NSKeyValueObservingOptions)options usingBlock:(KVOBlock)block {
+    KVOObservation *observation = [[KVOObservation alloc] initWithKeyPath:keyPath object:self block:block];
+    [self addObserver:observation forKeyPath:keyPath options:options context:NULL];
+    return observation;
+}
+
+- (void)observe:(NSArray<NSString *> *)keyPaths options:(NSKeyValueObservingOptions)options owner:(id)owner usingBlock:(void (^)(id self))block {
+    __weak id weakOwner = owner;
+    KVOBlock newBlock = ^{
+        id owner = weakOwner;
+        if (owner) {
+            block(owner);
+        }
```

Comment:
This sounds like a good time to disable early

Reply:
Actually, I would assert here, this should never happen…

Reply:
Done